src/shims/unix/unnamed_socket.rs

```diff
@@ -100,33 +104,31 @@ impl FileDescription for AnonSocket {
             // corresponding ErrorKind variant.
             throw_unsup_format!("reading from the write end of a pipe");
         };
+        let dest = dest.clone();
+        let weak_self_ref = self_ref.downgrade();
+
+        // We handle return earlier if we hit these two case, and only do the actual read and
+        // blocking in anonsocket_read.
```

Comment:
I don't understand this comment. What do you mean by "handle return"?

Reply:
Ah it should be ``"We return early..."``

But please ignore the comment for now, I plan to remove that because one of the cases should be moved to the helper function.